### PR TITLE
Switch to rollup

### PR DESCRIPTION
**Pull request by gkatsev**
This uses rollup to generate the following files:
* dist/vtt.js as a UMD file
* dist/vtt.global.js as an IIFE file
* dist/vtt.cjs.js as a CJS file
* dist/vtt.es.js as an ESM file
* dist/vtt.min.js as a UMD file
* dist/vtt.global.min.js as an IIFE file

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "presets": ["es2015"]
```
Comment (brandonocasey) [inline .babelrc:2]: should this have `{'loose': true, 'modules': false}`

File: rollup.js [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+
+const es = {
+  options: {
+    entry: 'lib/browser-index.js',
+    plugins: [
+      json(),
+      args.progress ? progress() : {},
+      filesize()
+    ],
+    onwarn(warning) {
```
Comment (brandonocasey) [inline rollup.js:46]: This could be nice in the generator :)

File: rollup.js [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+
+const minifiedGlobal = Object.assign({}, _.cloneDeep(umd), {
+  dest: 'dist/vtt.global.min.js'
+});
+
+minifiedGlobal.options.plugins.splice(4, 0, uglify({
+  preserveComments: 'some',
+  screwIE8: false,
+  mangle: true,
+  compress: {
```
Comment (brandonocasey) [inline rollup.js:119]: do we need all the specific options? are the same for every uglify here? If so we may just want to merge them all into a "defaults" object or something.

File: .babelrc (hunk 1 above)
Comment (gkatsev) [reply .babelrc:2]: No because it's only used to compile the `rollup.js` file. Babel in rollup is configured in `rollup.js`

Comment (gkatsev) [reply .babelrc:2]: and by that I mean we don't use ES6 syntax at all right now in the main code.

**Review by misteroneill — APPROVED** [review]

**Comment by gkatsev** [comment]
I give up with tests. Going to ignore them until we can upgrade them to a recent version of node.

**Comment by gkatsev** [comment]
Rollup is being finicky about the minified files. Going to switch tracks to deumdify instead.

**Pull request closed without merging**